src/Components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { usePopper } from 'react-popper';
+import { EllipsisButton } from './EllipsisButton';
+import { PopoverList, PopoverListItem } from './PopoverList';
+
+type ChartSettingsButtonProps = {
+    setShowAnnotation: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const ChartSettingsButton = ({ setShowAnnotation }: ChartSettingsButtonProps) => {
+    const [referenceElement, setReferenceElement] = useState(null);
+    const [popperElement, setPopperElement] = useState(null);
+
+    const [isVisible, setIsVisible] = useState(false);
+
+    const { styles, attributes } = usePopper(referenceElement, popperElement, {
+        placement: 'bottom',
+        modifiers: [
+            {
+                name: 'offset',
+                options: { offset: [0, 5] },
+            },
+        ],
+    });
+
+    return (
+        <div className='absolute top-2 right-2'>
+            <EllipsisButton onClick={() => setIsVisible(!isVisible)} isVisible={isVisible} setReferenceElement={setReferenceElement} />
```

Comment:
Use previous state to ensure that the state updates on the most recent state value, minimizing the risk of "hook desync".
```suggestion
            <EllipsisButton onClick={() => setIsVisible(prevState => !prevState)} isVisible={isVisible} setReferenceElement={setReferenceElement} />
```